torch/_export/serde/serialize.py

```diff
@@ -1488,6 +1524,105 @@ def deserialize_operator(self, serialized_target: str):
                 target = getattr(target, name)
         return target
 
+    def deserialize_sym_expr(self, s: SymExprNode, hint: Optional[int] = None, force_int: Optional[bool] = True) -> Union[sympy.Expr]:
+        if (base := s.base) is not None:
+            if base.type == "as_bool":
+                assert isinstance(base.value, bool)
+                return base.value
+            elif base.type == "as_float":
+                assert isinstance(base.value, float)
+                return base.value
+            elif base.type == "as_int":
+                assert isinstance(base.value, int)
+                return base.value
+            else:
+                assert base.type == "as_symbol"
+                assert isinstance(base.value, str)
+                if base.value in self.symbol_name_to_symbol:
+                    s = self.symbol_name_to_symbol[base.value]
+                else:
+                    s = sympy.Symbol(base.value, integer=force_int)
+                    self.symbol_name_to_symbol[base.value] = s
+                assert isinstance(s, sympy.Symbol)
+                if (
+                    hint is not None
+                    and not s in self.shape_env.var_to_val
+                ):
+                    self.shape_env.add_var_to_val(s, hint)
+                if vr := self.symbol_name_to_range.get(base.value):
+                    self.shape_env.constrain_symbol_range(
+                        s,
+                        compiler_min=vr.lower,  # type: ignore[arg-type]
+                        compiler_max=vr.upper,  # type: ignore[arg-type]
+                    )
+                return s
+        else:
+            assert s.target is not None
+            *_module, _qualname = s.target.split(".")
+            node_cls = getattr(sys.modules[".".join(_module)], _qualname)
+            expr = node_cls(*[self.deserialize_sym_expr(arg, force_int=force_int) for arg in s.args])
+            expr_str = str(expr)
+            if expr_str not in self.symbol_name_to_symbol:
+                self.symbol_name_to_symbol[expr_str] = expr
+            if vr := self.symbol_name_to_range.get(expr_str):
+                self.shape_env.constrain_symbol_range(
+                    expr,
+                    compiler_min=vr.lower,  # type: ignore[arg-type]
+                    compiler_max=vr.upper,  # type: ignore[arg-type]
+                )
+            return expr
+
+    def __deprecated_do_not_use_deserialize_symint_expr_str(self, s: str, hint: Optional[int] = None) -> sympy.Expr:
```

Comment:
do you think theres a way to implement this like an upgrader? If we see something using `expr_str` can we somehow change it to become an `expr_ast` and then use the `expr_ast` implementation? 

We eventually want to implement an upgrader/downgrader so if we just do that here then it could save us the work later.

Reply:
Yeahh this is a temporary fix while we figure out the upgrader design